app/src/main/java/com/jin/jjinweather/feature/weather/domain/model/Weather.kt

```diff
@@ -1,35 +1,67 @@
- * @property iconCode d 와 n는 day 와 night.
- * - 01d or 01n -> clear sky
- * - 02d or 02n -> few clouds
- * - 03d or 03n -> scattered clouds
- * - 04d or 04n -> broken clouds
- * - 09d or 09n -> shower rain
- * - 10d or 10n -> rain
- * - 11d or 11n -> thunderstorm
- * - 13d or 13n -> snow
- * - 50d or 50n -> mist
- * @property moonPhase 값의 범위는 0.0부터 1.0 까지 이며, 주요 값은 다음과 같다.
- * - daily 에서 첫번째 인덱스 값으로 적용
- * - 0.0 또는 1.0 -> New Moon(삭, 달이 태양과 일직선, 보이지 않음.)
- * - 0.25 -> First Quarter (상현달, 오늘쪽 반달)
- * - 0.5 -> Full Moon (보름달, 완전한 둥근달)
- * - 0.75 -> Last Quarter (하현달, 왼쪽 반달)
- */
+import androidx.annotation.DrawableRes
+import com.jin.jjinweather.R
+import java.time.Instant
+import java.time.LocalTime
+import java.util.Calendar
+
 data class Weather(
-    val iconCode: String,
-    val currentTemperature: Number,
-    val yesterdayTemperature: Number,
-    val minTemperature: Number,
-    val maxTemperature: Number,
-    val hourlyWeatherList: List<HourlyWeather>,
-    val dailyWeatherList: List<DailyWeather>,
-    val sunrise: Long,
-    val sunset: Long,
-    val moonPhase: Double
+    val dayWeather: DayWeather,
+    val yesterdayWeather: TemperatureSnapshot,
+    val forecast: Forecast
+)
+
+data class DayWeather(
+    val date: Calendar,
+    val icon: WeatherIcon,
+    val temperature: Number,
+    val sunrise: LocalTime,
+    val sunset: LocalTime,
+    val moonPhase: Double,
+    val temperatureRange: TemperatureRange
+)
+
+data class TemperatureSnapshot(
+    val timeStamp: Instant,
+    val icon: WeatherIcon,
+    val temperature: Number,
+)
+
+data class Forecast(
+    val hourly: HourlyForecast,
+    val daily: List<DailyForecast>
 )
+
+typealias HourlyForecast = List<TemperatureSnapshot>
+
+data class DailyForecast(
+    val date: Calendar,
+    val icon: WeatherIcon,
+    val temperatureRange: TemperatureRange
+)
+
+data class TemperatureRange(
+    val min: Number,
+    val max: Number
+)
+
+enum class WeatherIcon(@DrawableRes val drawableRes: Int, val codes: Set<String>) {
+    CLEAR_SKY_DAY(R.drawable.ic_main_clear_sky_day, setOf("01d")),
+    CLEAR_SKY_NIGHT(R.drawable.ic_main_clear_sky_night, setOf("01n")),
+    FEW_CLOUDS_DAY(R.drawable.ic_main_few_clouds_day, setOf("02d")),
+    FEW_CLOUDS_NIGHT(R.drawable.ic_main_few_clouds_night, setOf("02n")),
+    SCATTERED_CLOUDS(R.drawable.ic_main_scattered_clouds, setOf("03d", "03n", "04d", "04n")),
+    SHOWER_RAIN(R.drawable.ic_main_shower_rain, setOf("09d", "09n")),
+    RAIN_DAY(R.drawable.ic_main_rain_day, setOf("10d")),
+    RAIN_NIGHT(R.drawable.ic_main_rain_night, setOf("10n")),
+    THUNDERSTORM(R.drawable.ic_main_thunderstorm, setOf("11d", "11n")),
+    SNOW(R.drawable.ic_main_snow, setOf("13d", "13n")),
+    MIST(R.drawable.ic_main_mist, setOf("50d", "50n"));
+
+    companion object {
+        fun findByWeatherIcon(code: String): WeatherIcon {
+            return entries.firstOrNull { code in it.codes } ?: CLEAR_SKY_DAY
+        }
+    }
```

Comment:
```suggestion
    companion object {
        /**
         * A lookup table from the weather code to [WeatherIcon]
         */
        private val quickLookupTable: Map<String, WeatherIcon> =
            entries.fold(mutableMapOf()) { map, icon ->
                for (code in icon.codes) {
                    map[code] = icon
                }
                map
            }
    
        fun findByWeatherCode(code: String): WeatherIcon =
            map.getOrDefault(code, WeatherIcon.CLEAR_SKY_DAY)
    }
```

이런식으로 mapping table을 만드는 방식도 있습니다.
그렇지만, 지금 있는 linear search방식으로도 충분히 빠를거기 때문에 해제되지 않는 메모리 사용을 줄이기 위해서 mapping table을 유지하지 않는 방식도 충분히 유효할 것 같습니다.

외부에서 "codes"를 참조할 필요가 없다면, "codes" 멤버를 삭제하고,

```suggestion
        fun findByWeatherCode(code: String): WeatherIcon =
            when (code) {
              "01d" -> WeatherIcon.CLEAR_SKY_DAY
              ...
              else -> WeatherIcon.CLEAR_SKY_DAY
            }

```

도 괜찮은 선택입니다.

Reply:
넵 찾아보니 `mapping table` 방법은 호출 빈도가 엄청 많거나 맵핑 규칙이 자주 바뀌는 상황에서 사용시 적절하다고 확인했습니다. 기억 하였다가 필요시에 한번 사용하도록 하겠습니다!
또한, `codes` 멤버 삭제 후 when문으로 분기처리하도록 수정하였습니다.